src/NServiceBus.RavenDB/Subscriptions/SubscriptionPersister.cs

```diff
@@ -102,22 +102,34 @@ public SubscriptionPersister(IDocumentStore store)
                 await session.SaveChangesAsync().ConfigureAwait(false);
             }
         }
+    
 
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var ids = messageTypes.Select(Subscription.FormatId).ToList();
-
             using (var suppressTransaction = new TransactionScope(TransactionScopeOption.Suppress, TransactionScopeAsyncFlowOption.Enabled))
             {
                 Subscriber[] subscribers;
                 using (var session = OpenAsyncSession())
                 {
+                    var collectionName = session.Advanced.DocumentStore.Conventions.FindTypeTagName(typeof(Subscription));
```

Comment:
can this be moved in the constructor or in the feature initialization and passed in as an argument or there is value in checking every time since the user convention can mutate all times based on some external context we are unaware of?